docs/DeveloperGuide.adoc

```diff
@@ -554,6 +483,83 @@ Cell Formatting
+
+The implementation of the Statistics feature can be split into 2 phases, preparation and execution. Given below is an explanation of how the Statistics mechanism behaves at each phase.
+
+===== Preparation
+
+During the preparation phase, the program parses the command for Statistics and prepares to open a new Statistics Window.
+The following sequence diagram shows how `stats d/01/01/2019 d/07/01/2019` works during the preparation phase.
+
+.Sequence Diagram demonstrating the preparation stage of Statistics feature
+image::StatsSequenceDiagram.png[]
+
+NOTE: The lifeline for `StatsCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+1. User executes `stats d/01/01/2019 d/07/01/2019` to generate statistics for spending within `01/01/2019` and `07/01/2019`.
+
+2. SpendingBookParser will parse and identify the command as a StatsCommand and pass on the argument `d/01/01/2019 d/07/01/2019` to StatsCommandParser.
+
+3. StatsCommandParser will check whether the date range parameters provided are valid.
+
+4. `StatsCommand#execute()` is executed, returning a message which tells the user that the Stats Window will be opened and set a boolean `isShowStats` = true to indicate that a new window to display the statistics will be opened.
+
+===== Execution
+
+In the execution phase, the program processes and retrieves the data to be displayed and finally displaying it.
+
+There are 2 types of data that the program require, `statsMessage` for displaying the statistics in text and `statsData` for displaying the data in a piechart.
+Below is the UML sequence diagram and a step-by-step explanation for the retrieval of `statsData`. `statsMessage` is retrieved in the same manner.
+
+.Sequence Diagram of Execution Phase showing retrieving of `statsData`
+image::StatsDataSequenceDiagram.png[]
+
+
+NOTE: The lifeline for `StatsCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+
+1. Upon checking that the boolean `isShowStats` = true, the MainWindow calls `LogicManager#getStatsData()`.
+
+2. SpendingBookParser will parse and identify the command as a StatsCommand and pass on the argument `d/01/01/2019 d/07/01/2019` to StatsCommandParser.
+
+3. StatsCommandParser will check whether the date range parameters provided are valid.
+
+4. `StatsCommand#getStatsData()` is executed. Spending with date range within `01/01/2019` and `07/01/2019` is filtered.
+The filtering is done through a predicate parameter which checks through the dates of the spending in the application.
+
+5. Finally, the `statsData` will be displayed on the Ui through `StatsWindow#loadData()` call.
+
+==== Design Considerations
+
+===== How to pass statistics data parameters from Logic to UI
+
+====== Alternative 1 (current choice): UI gets all data and parameters from Logic, which gets data from Model.
+
```

Comment:
Are we using this format or a table format? to be consistent

Reply:
Table